src/NLog/Internal/FileAppenders/FileAppenderCache.cs

```diff
@@ -234,15 +234,14 @@ public void FlushAppenders()
                 appender.Flush();
             }
         }
-        
+
         /// <summary>
         /// Gets the file info for a particular appender.
         /// </summary>
         /// <param name="fileName">The file name associated with a particular appender.</param>
-        /// <param name="lastWriteTime">The last file write time. The value must be of UTC kind.</param>
-        /// <param name="fileLength">Length of the file.</param>
+        /// <param name="fileCharacteristics">The file characteristics, if the file information was retrieved successfully.</param>
         /// <returns><see langword="true"/> when the operation succeeded; <see langword="false"/> otherwise.</returns>
-        public bool GetFileInfo(string fileName, out DateTime lastWriteTime, out long fileLength)
+        public bool GetFileCharacteristics(string fileName, out FileCharacteristics fileCharacteristics)
```

Comment:
Looks good! But why it's still a out parameter? We can refactor the whole tree because it's internal?